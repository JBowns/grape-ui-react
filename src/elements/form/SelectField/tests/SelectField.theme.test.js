import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { SelectField } from '..';

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};
const colorOptions = [
  { color: '#FF5630', label: 'Red', value: 'red' },
  { color: '#FFC400', label: 'Yellow', value: 'yellow' },
  { color: '#36B37E', label: 'Green', value: 'green' },
];

describe('SelectField using Theme color', () => {
  it('should return a SelectField object with custom color', () => {
    const theme = {
      colors: {
        grapeSoda: {
          base: 'hsl(325, 84.6%, 28%)',
          dark: 'hsl(305, 33.9%, 23.7%)',
          light: 'hsl(313, 67.8%, 47.5%)',
        },
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <SelectField bg="grapeSoda.dark" color="grapeSoda.light" controlId="exampleControl" id="exampleControl" labelText="Color" options={colorOptions} sm value={colorOptions[1]} />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});

describe('SelectField using Theme Font Family', () => {
  it('should return a SelectField object with a custom font family', () => {
    const theme = {
      fonts: {
        trueSpace: '"DejaVu Sans Mono", "Bitstream Vera Sans Mono", Courier',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <SelectField
          controlId="exampleControl"
          fontFamily="trueSpace"
          id="exampleControl"
          labelText="Color"
          options={colorOptions}
          sm
          value={colorOptions[1]}
        />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});

describe('SelectField using custom brand colors', () => {
  it('should return a SelectField object with custom brand primary color', () => {
    const theme = {
      colors: {
        brandLink: {
          base: 'hsl(100, 84.6%, 28%)',
          dark: 'hsl(105, 33.9%, 23.7%)',
          light: 'hsl(113, 67.8%, 47.5%)',
        },
        brandLinkHover: {
          base: 'hsl(100, 84.6%, 38%)',
          dark: 'hsl(105, 33.9%, 33.7%)',
          light: 'hsl(113, 67.8%, 57.5%)',
        },
      },
      fonts: {
        base: 'Nunito, sans-serif',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <SelectField
          labelText="Color"
          name="selectColor"
          options={colorOptions}
          sm
          value={colorOptions[1]}
        />
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
