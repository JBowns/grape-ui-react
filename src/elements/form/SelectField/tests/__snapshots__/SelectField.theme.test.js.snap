// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField using Theme Font Family should return a SelectField object with a custom font family 1`] = `
.c2 {
  background-color: hsl(0,0%,100%);
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  height: 0.4rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  left: 8px;
  position: absolute;
  top: -1px;
  padding-left: 8px;
  padding-right: 8px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c0 {
  position: relative;
  background-color: transparent;
  color: inherit;
  padding-bottom: 16px;
  padding-top: 4px;
  width: 100%;
}

.c1 {
  background-color: transparent;
  color: inherit;
  font-family: "DejaVu Sans Mono","Bitstream Vera Sans Mono",Courier;
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  border-radius: 2px;
  outline: 0;
  width: 100%;
  padding: 1rem;
}

.c1 .grape-ui-select__control {
  margin: -1rem;
  padding: 1rem;
}

.c1[disabled] {
  opacity: 0.3;
}

.c1[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c1:focus,
.c1:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c1:focus + label,
.c1:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c1::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c1::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c1:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c1::placeholder {
  color: hsl(0,0%,75%);
}

<div
  className="c0"
>
  <div
    className="css-1pcexqc-container c1"
    id="exampleControl"
    onKeyDown={[Function]}
  >
    <div
      className="css-mazh0z grape-ui-select__control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-14cvlpc grape-ui-select__value-container grape-ui-select__value-container--has-value"
      >
        <div
          className="css-8uhtka grape-ui-select__single-value"
        >
          Yellow
        </div>
        <div
          className="css-0"
        >
          <div
            className="grape-ui-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-3-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1rtnzd2 grape-ui-select__indicators"
      >
        <span
          className="css-1hyfx7x grape-ui-select__indicator-separator"
        />
        <div
          aria-hidden="true"
          className="css-1pseabs-indicatorContainer grape-ui-select__indicator grape-ui-select__dropdown-indicator"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <label
    className="c2"
    htmlFor="exampleControl"
  >
    Color
  </label>
</div>
`;

exports[`SelectField using Theme color should return a SelectField object with custom color 1`] = `
.c2 {
  background-color: hsl(305,33.9%,23.7%);
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  height: 0.4rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  left: 8px;
  position: absolute;
  top: -1px;
  padding-left: 8px;
  padding-right: 8px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c0 {
  position: relative;
  background-color: hsl(305,33.9%,23.7%);
  color: inherit;
  padding-bottom: 16px;
  padding-top: 4px;
  width: 100%;
}

.c1 {
  background-color: hsl(305,33.9%,23.7%);
  color: hsl(313,67.8%,47.5%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  border-radius: 2px;
  outline: 0;
  width: 100%;
  padding: 1rem;
}

.c1 .grape-ui-select__control {
  margin: -1rem;
  padding: 1rem;
}

.c1[disabled] {
  opacity: 0.3;
}

.c1[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c1:focus,
.c1:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c1:focus + label,
.c1:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c1::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c1::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c1:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c1::placeholder {
  color: hsl(0,0%,75%);
}

<div
  className="c0"
>
  <div
    className="css-1pcexqc-container c1"
    id="exampleControl"
    onKeyDown={[Function]}
  >
    <div
      className="css-mazh0z grape-ui-select__control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-14cvlpc grape-ui-select__value-container grape-ui-select__value-container--has-value"
      >
        <div
          className="css-8uhtka grape-ui-select__single-value"
        >
          Yellow
        </div>
        <div
          className="css-0"
        >
          <div
            className="grape-ui-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1rtnzd2 grape-ui-select__indicators"
      >
        <span
          className="css-1hyfx7x grape-ui-select__indicator-separator"
        />
        <div
          aria-hidden="true"
          className="css-1pseabs-indicatorContainer grape-ui-select__indicator grape-ui-select__dropdown-indicator"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <label
    className="c2"
    htmlFor="exampleControl"
  >
    Color
  </label>
</div>
`;

exports[`SelectField using custom brand colors should return a SelectField object with custom brand primary color 1`] = `
.c2 {
  background-color: hsl(0,0%,100%);
  color: inherit;
  font-family: Nunito,sans-serif;
  font-size: 0.8rem;
  font-weight: inherit;
  height: 0.4rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  left: 8px;
  position: absolute;
  top: -1px;
  padding-left: 8px;
  padding-right: 8px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c0 {
  position: relative;
  background-color: transparent;
  color: inherit;
  padding-bottom: 16px;
  padding-top: 4px;
  width: 100%;
}

.c1 {
  background-color: transparent;
  color: inherit;
  font-family: Nunito,sans-serif;
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  border-radius: 2px;
  outline: 0;
  width: 100%;
  padding: 1rem;
}

.c1 .grape-ui-select__control {
  margin: -1rem;
  padding: 1rem;
}

.c1[disabled] {
  opacity: 0.3;
}

.c1[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c1:focus,
.c1:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c1:focus + label,
.c1:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c1::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c1::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c1:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c1::placeholder {
  color: hsl(0,0%,75%);
}

<div
  className="c0"
>
  <div
    className="css-1pcexqc-container c1"
    id="selectColor"
    onKeyDown={[Function]}
  >
    <div
      className="css-mazh0z grape-ui-select__control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-14cvlpc grape-ui-select__value-container grape-ui-select__value-container--has-value"
      >
        <div
          className="css-8uhtka grape-ui-select__single-value"
        >
          Yellow
        </div>
        <div
          className="css-0"
        >
          <div
            className="grape-ui-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-4-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1rtnzd2 grape-ui-select__indicators"
      >
        <span
          className="css-1hyfx7x grape-ui-select__indicator-separator"
        />
        <div
          aria-hidden="true"
          className="css-1pseabs-indicatorContainer grape-ui-select__indicator grape-ui-select__dropdown-indicator"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="selectColor"
      type="hidden"
      value="yellow"
    />
  </div>
  <label
    className="c2"
    htmlFor="selectColor"
  >
    Color
  </label>
</div>
`;
